sql/thriftserver/if/TCLIService.thrift

```diff
@@ -0,0 +1,1298 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Coding Conventions for this file:
+//
+// Structs/Enums/Unions
+// * Struct, Enum, and Union names begin with a "T",
+//   and use a capital letter for each new word, with no underscores.
+// * All fields should be declared as either optional or required.
+//
+// Functions
+// * Function names start with a capital letter and have a capital letter for
+//   each new word, with no underscores.
+// * Each function should take exactly one parameter, named TFunctionNameReq,
+//   and should return either void or TFunctionNameResp. This convention allows
+//   incremental updates.
+//
+// Services
+// * Service names begin with the letter "T", use a capital letter for each
+//   new word (with no underscores), and end with the word "Service".
+
+namespace java org.apache.spark.sql.thriftserver.cli.thrift
+namespace cpp apache.spark.sql.thriftserver.cli.thrift
+
+// List of protocol versions. A new token should be
+// added to the end of this list every time a change is made.
```

Comment:
@juliuszsompolski Ok with this package name?

Reply:
:+1:

Reply:
> 👍

New code in https://github.com/AngersZhuuuu/spark/tree/SPARK-29018-V11-STEP2

Current change list:
1. Impelemnt `Type` in scala since spark con't support all type of hive
2. Implement `Service/AbstractService` prepare for remove hive conf in future
3. Construct RowSet with StructType and Row
4. Implement `HiveAuthFactory` since between 1.2.1/2.3.5, their delegation token managerment changed. Impelment on DelegationTokenMnagerment by scala
5. MV `tableTypeString` from `SparkMetadataOperationUtils` to `SparkMetadataOperation`
5. Since there are `tableTypeString` in `SparkMetadataOperation` remove `ClassicTypeMapping`, `HiveTableTypeMapping`, `TableTypeMapping` and `TableTypeMappingFactory`
6. Implement all operation for spark since it execute in different way
7. Add new method `GetQueryId`, `SetClientInfo` for thrift version v11 in `ThriftCLIService`
8. Add `statementid` to `Operation` for implement `GetQueryId`
9. Remove `GlobalHivercFileProcessor` `setFetchSize` `processGlobalInitFile`etc


Still working on this.